app/controllers/signup_controller.rb

```diff
@@ -88,17 +85,20 @@ def profile
       handle_with(handler,
                   contracts_required: !contracts_not_required,
                   success: lambda do
-                    clear_signup_state
-
-                    if current_user.student?
+                    if current_user.student? || current_user.created_from_trusted_data?
                       redirect_back
                     else
                       redirect_to action: :instructor_access_pending
                     end
+                    clear_signup_state
```

Comment:
why does it matter if `clear_signup_state` is called here or at the start of the block?  Will happen either way, and the redirects don't depend on it?

Reply:
I'm guessing a previous incarnation of the code read `signup_state` when deciding where redirect.  I'll move it back